src/metric/definition.go

```diff
@@ -66,10 +66,22 @@ var APIServerSpecs = definition.SpecGroups{
 				),
 				Type: sdkMetric.RATE,
 			},
+			// etcd_object_counts was deprecated in k8s 1.22 and removed in 1.23 (it is replaced by apiserver_storage_objects)
 			{
 				Name:      "etcdObjectCounts",
 				ValueFunc: prometheus.FromValueWithOverriddenName("etcd_object_counts", "etcdObjectCounts"),
 				Type:      sdkMetric.GAUGE,
+				Optional:  true,
+			},
+			// apiserver_storage_objects was introduced in k8s 1.21 and replaces etcd_object_counts in 1.23
+			{
+				Name: "apiserverStorageObjects",
+				ValueFunc: fetchIfMissing(
+					prometheus.FromValueWithOverriddenName("apiserver_storage_objects", "apiserverStorageObjects"),
+					prometheus.FromValueWithOverriddenName("etcd_object_counts", "etcdObjectCounts"),
+				),
+				Type:     sdkMetric.GAUGE,
+				Optional: true,
```

Comment:
Should it be optional though? `etcdObjectCounts` wasn't, so shouldn't this metric be always present, due to the `fetchIfMissing` logic?

Reply:
I think you are right... 🤔  I guess I kept it as optional because the first draft of `fetchIfMissing` returned an error instead of an `empty.FetchedValues{}` when the metric to be replaced was not missing. I'm double-checking if everything is taken into account and removing the optional flag. Thanks!

Reply:
That was it, thanks again! Updated here: dee1a3e3b9ea4285d8896fc79a49bbaf9ff6f975